pkg/engine/imageVerify.go

```diff
@@ -355,6 +355,44 @@ func (iv *imageVerifier) verifyAttestorSet(attestorSet kyvernov1.AttestorSet, im
 	return nil, err
 }
 
+func expandKeysData(attestorSet kyvernov1.AttestorSet) kyvernov1.AttestorSet {
+	attestorSet = expandStaticKeys(attestorSet)
+	attestorSet = expandSecretData(attestorSet)
+	attestorSet = expandKmsData(attestorSet)
```

Comment:
Why do we need to `expand` the secret and KMS entries?

Reply:
in `expandStaticKeys`, we are setting `attestor.entries` to empty list if public keys are not present. if we are passing secret or kms, we need to consider those. hence wrote similar function to add them in `attestor.entries`. function name can be change though

Reply:
changed function name to `extractSecret` and `extractKms`

Reply:
Hi @shahpratikr - thanks for the changes, I now understand why you had to add the other key types. 

However, this method is still a bit confusing to me. I realize its minor, but it would be good to clean up to make it more understandable and maintainable in the future.

The goal of `expandStaticKeys` was to allow users to specify PEM delimited public keys that Kyverno internally expands into separate attestor entries within the attestor set, and applies a logical AND operation. 

Can we combine these 3 methods into a single one that does this and returns the modified attestorSet?

And would be good to add a comment to the method so we can easily understand its purpose in the future.

And, lets add a few unit test cases to cover handling of the different key types with PEM delimited data.

Reply:
Hi @JimBugwadia, it turns out we don't need to make any changes to add secret and kms separately into `attestor.Entries`. [Line](https://github.com/kyverno/kyverno/blob/main/pkg/engine/imageVerify.go#L370) adds both in entries.
I tried to add unit test similar to e2e tests, but the issue with unit test is we cannot mock kubernetes client used to get public key from secrets. [Line](https://github.com/kyverno/kyverno/blob/main/pkg/cosign/cosign.go#L151) calls third party package which provides us with public key detail. Please let me know if you have any work around for this situation, I'll be happy to implement that for unit tests.

Reply:
@JimBugwadia can you please reply?